modules/nodes/main.tf

```diff
@@ -66,7 +66,7 @@ resource "terraform_data" "calico_setup" {
   count = try(fileexists(var.eks_info.k8s_pre_setup_sh_file), false) ? 1 : 0
 
   triggers_replace = [
-    filemd5(var.eks_info.k8s_pre_setup_sh_file)
+    var.eks_info.k8s_pre_setup_change_hash
```

Comment:
I would add an entry in `modules/eks/submodules/k8s/templates/k8s-functions.sh.tftpl`
`K8S_function_hash=md5(k8s-functions.sh`

then keep the trigger as `filemd5(var.eks_info.k8s_pre_setup_sh_file)`
I believe it would avoid a new var being passed around

Reply:
ah yeah, nice!